src/main/java/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,18 @@
+package strategy;
+
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> inventory, String fruit, int quantity) {
+        int currentQuantity = inventory.getOrDefault(fruit, 0);
+        if (currentQuantity < quantity) {
+            throw new RuntimeException("Not enough " + fruit + " in inventory.");
+        }
```

Comment:
This check is redundant because the subsequent check (newQuantity < 0) will already cover the case where currentQuantity is less than quantity. You can remove this to simplify the code.